dll/win32/shell32/droptargets/CFSDropTarget.cpp

```diff
@@ -83,12 +83,27 @@ HRESULT CFSDropTarget::_CopyItems(IShellFolder * pSFFrom, UINT cidl,
     if (!pszSrcList)
         return E_OUTOFMEMORY;
 
+    BOOL bRenameOnCollision = FALSE;
+    if (bCopy)
+    {
+        /* Source file might be one or more */
+        WCHAR szPath1[MAX_PATH], szPath2[MAX_PATH];
+        GetFullPathNameW(pszSrcList, _countof(szPath1), szPath1, NULL);
+        GetFullPathNameW(m_sPathTarget, _countof(szPath2), szPath2, NULL);
+        PathRemoveFileSpecW(szPath1);
+        PathRemoveFileSpecW(szPath2);
+        if (_wcsicmp(szPath1, szPath2) == 0)
+            bRenameOnCollision = TRUE;
+    }
+
     SHFILEOPSTRUCTW op = {0};
     op.pFrom = pszSrcList;
     op.pTo = wszTargetPath;
     op.hwnd = m_hwndSite;
     op.wFunc = bCopy ? FO_COPY : FO_MOVE;
     op.fFlags = FOF_ALLOWUNDO | FOF_NOCONFIRMMKDIR;
+    if (bRenameOnCollision)
```

Comment:
The question from above would be equivalent to removing the whole added code block from above, and replacing this condition by just `if (bCopy)  op.fFlags |= FOF_RENAMEONCOLLISION;` .

Reply:
Copying from another directory shouldn't rename automatically. It should confirm overwrite.